src/backend/distributed/executor/executor.c

```diff
@@ -2,6 +2,86 @@
  *
  * executor.c
  *
+ * The adaptive executor executes a list of tasks (queries on shards) over
```

Comment:
In general, I really like how you described the executor here. I've some notes/questions:

- Given that I'm very familiar with the code, I can quickly follow the text here. Do you think would it make sense to get a quick review from someone else as well?
- Should we consider renaming `executor.c` to `adaptive_executor.c`. We'll at least have the task tracker for a while :) 
- Should we very briefly mention about the unusual interpretation of `SEQUENTIAL_CONNECTION` vs `SequentialRunDistributedExecution`? (Maybe only even refer to `ShouldRunTasksSequentially` function where we explain this to some extent)
- Would it make sense to mention about the failure handling here? We're handling failures differently for some cases compared to the other executions, and we probably have a very little comment that explains failure handling in general.

Reply:
>     * Should we very briefly mention about the unusual interpretation of `SEQUENTIAL_CONNECTION` vs `SequentialRunDistributedExecution`? (Maybe only even refer to `ShouldRunTasksSequentially` function where we explain this to some extent)

@onderkalaci What do you think about renaming `SEQUENTIAL_CONNECTION` to `SINGLE_CONNECTION_PER_WORKER` in another PR so it doesn't need explanation?

Reply:
Added some note on sequential execution, failure handling, and did the rename.